pkg/translator/prometheusremotewrite/metrics_to_prw_v2.go

```diff
@@ -134,30 +138,54 @@ func (c *prometheusConverterV2) fromMetrics(md pmetric.Metrics, settings Setting
 
 // timeSeries returns a slice of the writev2.TimeSeries that were converted from OTel format.
 func (c *prometheusConverterV2) timeSeries() []writev2.TimeSeries {
-	allTS := make([]writev2.TimeSeries, 0, len(c.unique))
+	allTS := make([]writev2.TimeSeries, 0, len(c.unique)+c.conflictCount)
 	for _, ts := range c.unique {
 		allTS = append(allTS, *ts)
 	}
+	for _, cTS := range c.conflicts {
+		for _, ts := range cTS {
+			allTS = append(allTS, *ts)
+		}
+	}
 	return allTS
 }
 
 func (c *prometheusConverterV2) addSample(sample *writev2.Sample, lbls []prompb.Label, metadata metadata) {
-	// TODO consider how to accommodate metadata in the symbol table when allocating the buffer, given not all metrics might have metadata.
-	buf := make([]uint32, 0, len(lbls)*2)
+	ts, isNewMetric := c.getOrCreateTimeSeries(lbls, metadata, sample)
+	// If the time series is not new, we can just append the sample to the existing time series.
+	if !isNewMetric {
+		ts.Samples = append(ts.Samples, *sample)
+	}
+}
+
+// isSameMetricV2 checks if two time series are the same metric
+func isSameMetricV2(ts1, ts2 *writev2.TimeSeries) bool {
+	if len(ts1.LabelsRefs) != len(ts2.LabelsRefs) {
+		return false
+	}
+	// As the labels are ordered as name, value, name, value, ... we can compare the labels by index jumping 2 steps at a time
+	for i := 0; i < len(ts1.LabelsRefs); i += 2 {
+		if ts1.LabelsRefs[i] != ts2.LabelsRefs[i] || ts1.LabelsRefs[i+1] != ts2.LabelsRefs[i+1] {
```

Comment:
```suggestion
	for i := 0; i < len(ts1.LabelsRefs); i++ {
		if ts1.LabelsRefs[i] != ts2.LabelsRefs[i] {
```
This is simpler, is equivalent, and it fixes a hypotheical out-of-bounds panic (if there was an odd number of labels somehow).

Reply:
[b5ca43c](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/39570/commits/b5ca43ca6046b724ad8a8187043b75dcec2b2b83)

Thanks. I did it here.